cfme/infrastructure/host.py

```diff
@@ -607,29 +608,25 @@ def find_quadicon(host, do_not_navigate=False):
     """Find and return a quadicon belonging to a specific host
 
     Args:
-        host: Host name as displayed at the quadicon
+        host: Host object
     Returns: :py:class:`cfme.web_ui.Quadicon` instance
     """
     if not do_not_navigate:
-        navigate_to(Host, 'All')
+        navigate_to(host, 'All')
     for page in paginator.pages():
-        quadicon = Quadicon(host, "host")
+        quadicon = Quadicon(host.name, "host")
         if sel.is_displayed(quadicon):
             return quadicon
     else:
-        raise HostNotFound("Host '{}' not found in UI!".format(host))
+        raise HostNotFound("Host '{}' not found in UI!".format(host.name))
 
 
-def navigate_and_select_all_hosts(host_names, provider_crud=None):
+def navigate_and_select_all_hosts(hosts):
     """ Reduces some redundant code shared between methods """
-    if isinstance(host_names, basestring):
-        host_names = [host_names]
-
-    if provider_crud:
-        navigate_to(provider_crud, 'ProviderNodes')
-    else:
-        navigate_to(Host, 'All')
+    if not isinstance(hosts, list):
+        hosts = [hosts]
+    navigate_to(hosts[0], 'All')
```

Comment:
It will not work for Openstack, because it demands now `isinstance(self.obj.provider, OpenstackInfraProvider)` inside the navigation class!